src/Contexts/user-context.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useReducer, createContext, Dispatch, useContext } from "react";
+
+import { UserInfo } from "../Types/User.interface";
```

Comment:
files and folder names should be in camel case.  Instead of Types it should be types